server.spi/src/main/java/org/kaazing/gateway/server/spi/security/AccessControlLoginException.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2007-2014 Kaazing Corporation. All rights reserved.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.kaazing.gateway.server.spi.security;
+
+import javax.security.auth.login.LoginException;
+
+/**
+ * Exception thrown from {@link javax.security.auth.spi.LoginModule} implementation
+ * for Gateway to react appropriately.
+ *
+ */
+public class AccessControlLoginException extends LoginException {
+    private static final long serialVersionUID = 1595142940282323729L;
+
+    public AccessControlLoginException() {
+        super();
+    }
+
+    public AccessControlLoginException(String message) {
+        super(message);
+    }
+}
```

Comment:
I'd like to understand the need for a custom exception, as ideally it would not be required.